packages/@ember/-internals/glimmer/lib/helpers/-disallow-dynamic-resolution.ts

```diff
@@ -0,0 +1,57 @@
+/**
+@module ember
+*/
+import { assert } from '@ember/debug';
+import { CapturedArguments } from '@glimmer/interfaces';
+import { createComputeRef, valueForRef } from '@glimmer/reference';
+import { internalHelper } from './internal-helper';
+
+export default internalHelper(({ positional, named }: CapturedArguments) => {
+  assert(
+    `[BUG] wrong number of positional arguments, expecting 1, got ${positional.length}`,
+    positional.length === 1
+  );
+
+  let nameOrValueRef = positional[0];
+
+  assert(`[BUG] expecting \`type\` named argument`, 'type' in named);
+  assert(`[BUG] expecting \`loc\` named argument`, 'loc' in named);
+  assert(`[BUG] expecting \`original\` named argument`, 'original' in named);
+
+  let typeRef = named.type;
+  let locRef = named.loc;
+  let originalRef = named.original;
+
+  // Bug: why do these fail?
+  // assert('[BUG] expecting a string literal for the `type` argument', isConstRef(typeRef));
+  // assert('[BUG] expecting a string literal for the `loc` argument', isConstRef(locRef));
+  // assert('[BUG] expecting a string literal for the `original` argument', isConstRef(originalRef));
```

Comment:
@pzuraq I don't think this has to block merging but it seems odd that we lost the const reference propagation here:

```hbs
{{helper (-disallow-dynamic-resolution this.helperName type="helper" loc=" (some-file.hbs:1:2) " original="this.helperName") "helper args"}}
                                                            ~~~~~~~~     ~~~~~~~~~~~~~~~~~~~~~~~          ~~~~~~~~~~~~~~~~~
```

It seemed strange that `isConstRef` returned false on this. The logic in `isConstRef` checks for `ref.tag === CONST_TAG` but `ref.tag` is `null` here. Seems like a bug?